napari/_vispy/vispy_points_layer.py

```diff
@@ -123,3 +128,54 @@ def _on_highlight_change(self, event=None):
             )
 
         self.node.update()
+
+    def _on_text_change(self, update_node=True):
+        """Function to update the text node properties
+
+        Parameters
+        ----------
+        update_node : bool
+            If true, update the node after setting the properties
+        """
+        ndisplay = self.layer.dims.ndisplay
+        if (len(self.layer._indices_view) == 0) or (
+            self.layer._text.visible is False
+        ):
+            text_coords = np.zeros((1, ndisplay))
+            text = []
+            anchor_x = 'center'
+            anchor_y = 'center'
+        else:
+            text_coords, anchor_x, anchor_y = self.layer._view_text_coords
+            if len(text_coords) == 0:
+                text_coords = np.zeros((1, ndisplay))
+            text = self.layer._view_text
+        text_node = self._get_text_node()
+        update_text(
+            text_values=text,
+            coords=text_coords,
+            anchor=(anchor_x, anchor_y),
+            rotation=self.layer._text.rotation,
+            color=self.layer._text.color,
+            size=self.layer._text.size,
+            ndisplay=ndisplay,
+            text_node=text_node,
+        )
+
+        if update_node:
+            self.node.update()
+
+    def _get_text_node(self):
+        """Function to get the text node from the Compound visual"""
+        text_node = self.node._subvisuals[-1]
+        return text_node
+
+    def _on_blending_change(self, event=None):
+        """Function to set the blending mode"""
+        self.node.set_gl_state(self.layer.blending)
+
+        # the text blending mode should always be additive
+        # see: https://github.com/napari/napari/pull/600#issuecomment-554142225
+        text_node = self._get_text_node()
+        text_node.set_gl_state('additive')
```

Comment:
This bit!!!!

... Aaaand I just saw that the comment linked above was mine! 😂 BUT, I'll emphasise one thing I said in the linked comment:

> I'm not entirely sure of all the consequences, though...

Now we know!!! What do people think? I mean, certainly default black text with additive blending is the worst of all worlds. 😂

Reply:
Ah, so you have met past Juan ;) I agree the default of black was a bad choice on my part. I think we have a couple of options:

1. we can add an blending parameter (with default) to `TextManager` that is set independently of the layer blending mode.

2. we can keep the blending mode as always additive, but change the default to something more sensible (e.g., cyan as you suggested).

I am leaning towards option (1) since it will be functionally the same as option (2) for the vast majority of users, but give some flexibility for "power users".

Reply:
To address this I have:

- added `TextManager.blending`
- `TextManager.blending` cannot be set to opaque. If the user attempts to set it to opaque, napari emits a warning and instead falls back to translucent.